e2e/errors.go

```diff
@@ -27,6 +29,9 @@ func isRetryableAPIError(err error) bool {
 		apierrors.IsTooManyRequests(err) || utilnet.IsProbableEOF(err) || utilnet.IsConnectionReset(err) {
 		return true
 	}
+	if strings.Contains(err.Error(), "etcdserver: request timed out") {
```

Comment:
Is this really needed? I would expect that this matches `apierrors.IsTimeout(err)`. In case it is needed, a comment that `IsTimeout()` does not match would be suitable.

Reply:
I didn't try. looks like it was not working as expected for the same reason we added this case at https://github.com/ceph/ceph-csi/blob/devel/e2e/pod.go#L85. am moving the check to `isRetryableAPIError` so that this can easily be reused for other calls also.

Reply:
I have removed this one hoping this will be covered by either `apierrors.IsTimeout(err)` or `apierrors.IsServerTimeout(err)` if that also does not helps we can bring it back.